internal/app/machined/pkg/controllers/network/cmdline.go

```diff
@@ -265,12 +265,11 @@ func ParseCmdlineNetwork(cmdline *procfs.Cmdline) (CmdlineNetworking, error) {
 			return settings, fmt.Errorf("malformed vlan commandline argument: %s", *vlanSettings)
 		}
 
-		numStart := strings.LastIndexFunc(vlanName, func(c rune) bool { return c >= '0' && c <= '9' })
```

Comment:
just changing `LastIndexFunc` to `IndexFunc` should be sufficient I believe

Reply:
Format is like "eth0:169" or something like that. Won't IndexFunc find the 0 in eth0? I just tested and test fails with NetworkLinkSpecs: ([]network.LinkSpecSpec) <nil>